src/Extractor.cc

```diff
@@ -3382,6 +3448,20 @@ namespace insur {
     return res.str();
   }
 
+ std::string Extractor::arbitraryLengthVector(std::string name, std::vector<double> invec){
```

Comment:
- Vector copy:
Whenever you see a vector, please make sure you are not copying it by value when not needed.
Here this vector is only read: **you should pass it by const reference**.

- Variable names:
Nice to be speicfic about names: for example here, what is it the name of? You could use moduleName, ladderName, parentVolumeName, etc. 
Same for invec: ideally, the names should already tell you what the variable represent. Instead, one should not repeat the type name: it is clear that elements in a vector, are in a vector. Rather use: modulesInRingTranslations, or modulesInRingRotations, or modulesInRingPositions, etc.